uast/transformer/positioner/positions.go

```diff
@@ -70,19 +93,77 @@ func fillLineColFromOffset(idx *positionIndex, pos *uast.Position) error {
 	return nil
 }
 
+func fromUnicodeOffset(idx *positionIndex, pos *uast.Position) error {
+	off, err := idx.RuneOffset(int(pos.Offset))
+	if err != nil {
+		return err
+	}
+	pos.Offset = uint32(off)
+	return fromOffset(idx, pos)
+}
+
+// runeSpan represents a sequence of UTF8 characters of the same size in bytes.
+type runeSpan struct {
+	runeOff  int // index of the first rune
```

Comment:
If this is an index, the name probably shouldn't imply "offset". Maybe: `firstRuneIndex`? I assume this is meant to indicate the _n_th rune from the beginning of the input?